### PR TITLE
Fix types over memoization & migrate interface Event to ICalendarEvent<T>

### PR DESCRIPTION
**Pull request by sebastienfi**
This address an issue with typings which were not making it through memoization.
Migrate the type `Event<T>` to `ICalendarEvent<T>` to avoid type collision with native `Event` [type](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.event.html).

### CODE REVIEW

**Comment by sebastienfi** [comment]
@acro5piano Don't know why the tests fails. I've tested under Windows and REHL 7 with no problem. Can you re-run it?

File: README.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ yarn add -D @types/react @types/react-native
 # Getting Started
 
 ```typescript
-import { Calendar } from 'react-native-big-calendar'
+import Calendar from 'react-native-big-calendar'
```
Comment (acro5piano) [inline README.md:43]: Can you restore it? I prefer named export to default export.

**Review by acro5piano — CHANGES_REQUESTED** [review]
The diff is huge so let me check it later more.

**Comment by acro5piano** [comment]
There seems to be a problem on our unit test. The result sometimes changes.

File: README.md (hunk 1 above)
Comment (sebastienfi) [reply README.md]: Sure! 
I've added both exports so it fits every style and updated documentation so cherry-picking is proposed as the first option.

```
import { Calendar } from './Calendar'
export * from './Calendar'
export * from './interfaces'
export default Calendar
```

**Comment by sebastienfi** [comment]
> There seems to be a problem on our unit test. The result sometimes changes.

Tests pass every time locally. Might it be a CI problem?

image

**Comment by acro5piano** [comment]
> > There seems to be a problem on our unit test. The result sometimes changes.
> 
> Tests pass every time locally. Might it be a CI problem?
> 
> image

Yeah, it should be fixed, but for now rerunning test in CI might fix it.

Thanks for your contribution and I'll check it tomorrow and hopefully release a new version!

File: README.md (hunk 1 above)
Comment (acro5piano) [reply README.md]: That's great!

**Comment by acro5piano** [comment]
@sebastienfi Sorry for the delay and thank you!

**Pull request merged**